src/components/filters/__tests__/checkbox.test.tsx

```diff
@@ -0,0 +1,113 @@
+import React from "react"
+import { fireEvent, render } from "@testing-library/react"
+
+import CheckboxFilter from "../checkbox"
+import ClockOutlinedTealIcon from "../../icons/clockOutlinedTeal"
+
+const renderWrapper = ({
+  name = "testFilter",
+  options = [
+    { label: "One", value: 1, renderIcon: null },
+    { label: "Two", value: 2, renderIcon: null },
+    { label: "Three", value: 3, renderIcon: null },
+  ],
+  applyFilters = jest.fn(),
+  selected = [],
+  facet = {},
+  children = null,
+} = {}) => {
+  return render(
+    <CheckboxFilter
+      name={name}
+      options={options}
+      applyFilters={applyFilters}
+      selected={selected}
+      facet={facet}
+    >
+      {children}
+    </CheckboxFilter>
+  )
+}
+
+describe("<CheckboxFilter/>", () => {
+  describe("renders correctly", () => {
+    it("with no options", () => {
+      const { container } = renderWrapper({ options: [] })
+
+      expect(container).toMatchSnapshot()
+    })
+
+    it("with simple options", () => {
+      const { container } = renderWrapper({
+        options: [
+          { label: "One", value: 1, renderIcon: null },
+          { label: "Two", value: 2, renderIcon: null },
+          { label: "Three", value: 3, renderIcon: null },
+        ],
+      })
+
+      expect(container).toMatchSnapshot()
+    })
+
+    it("with icons", () => {
+      const { container } = renderWrapper({
+        options: [
+          {
+            label: "One",
+            value: 1,
+            renderIcon: () => (
+              <ClockOutlinedTealIcon width="20" height="20" className="mr-10" />
+            ),
+          },
+          {
+            label: "Two",
+            value: 2,
+            renderIcon: () => (
+              <ClockOutlinedTealIcon width="20" height="20" className="mr-10" />
+            ),
+          },
+          {
+            label: "Three",
+            value: 3,
+            renderIcon: () => (
+              <ClockOutlinedTealIcon width="20" height="20" className="mr-10" />
+            ),
+          },
+        ],
+      })
+
+      expect(container).toMatchSnapshot()
+    })
+
+    it("with initial selection", () => {
+      const { container } = renderWrapper({ selected: [1, 2] })
+
+      expect(container).toMatchSnapshot()
+    })
+
+    it("with facet", () => {
+      const { container } = renderWrapper({
+        facet: { "1": 30, "2": 1000, "3": 0 },
+      })
+
+      expect(container).toMatchSnapshot()
+    })
+
+    it("with title", () => {
+      const { container } = renderWrapper({
+        children: <div>Title</div>,
+      })
+
+      expect(container).toMatchSnapshot()
```

Comment:
I would prefer to assert on the title being present over using snapshots allover the place :)

Reply:
Makes sense, I was lazy and copied existing tests